### DISCUSSION TITLE
我如何动动小手就让 CI 时间减少了 10 分钟 - XX’s Blog

### DISCUSSION THREAD

**Discussion by giscus** [xxchan/xxchan.github.io#39] [Announcements]
# 我如何动动小手就让 CI 时间减少了 10 分钟 - XX’s Blog

Stupidly effective ways to optimize Rust compile time

https://xxchan.github.io/cs/2023/02/11/optimize-rust-comptime.html

**Comment by dobzhao** [comment]
学到了很多，另外我也分享一个点，我之前尝试在Docker镜像里编译，Dockerfile里先ADD Cargo.toml进去，然后用echo随便生成个src/main.rs，cargo build一次编译所有依赖，然后再真正ADD项目源码进去，再build一次，这样做，之后不改Cargo.toml时docker build就会跳过第一次对所有依赖的编译步骤，直接找到依赖编译后的layer往下走，当然这个方法有效的前提是CI（虚拟）机不是每次构建才创建的。

**Reply by xxchan** [reply]
关于在 docker 里编译 rust 项目，有个 cargo-chef 好像是解决这个问题的可以看下（我还没试过
